File: src/main/java/yunas/exception/YunasException.java
package yunas.exception;

/**
 * YunasApp Exception.
 *
 *
 * @author Yosuke Kobayashi
 */
public class YunasException extends RuntimeException {

    private String code = null;


    YunasException(String code, String message)  {
        super(message);
        this.code = code;
    }

    YunasException(String code, String message, Throwable cause)  {
        super(message, cause);
        this.code = code;
    }
}

File: src/main/kotlin/yunas/exception/YunasException.kt
package yunas.exception

/**
 * YunasApp Exception.
 *
 *
 * @author Yosuke Kobayashi
 */
class YunasException : RuntimeException {

    private var code: String? = null


    internal constructor(code: String, message: String) : super(message) {
        this.code = code
    }

    internal constructor(code: String, message: String, cause: Throwable) : super(message, cause) {
        this.code = code
    }
}
